internal/crypto/cms/encoding/ber/ber.go

```diff
@@ -0,0 +1,301 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ber decodes BER-encoded ASN.1 data structures and encodes in DER.
+// Note:
+//   - DER is a subset of BER.
+//   - Indefinite length is not supported.
+//   - The length of the encoded data must fit the memory space of the int type
+//     (4 bytes).
+//
+// Reference:
+// - http://luca.ntop.org/Teaching/Appunti/asn1.html
+// - ISO/IEC 8825-1:2021
+package ber
+
+import (
+	"bytes"
+	"encoding/asn1"
+	"fmt"
+)
+
+// value is the interface for an ASN.1 value node.
+type value interface {
+	// EncodeMetadata encodes the identifier and length in DER to the buffer.
+	EncodeMetadata(*bytes.Buffer) error
+
+	// EncodedLen returns the length in bytes of the encoded data.
+	EncodedLen() int
+
+	// Content returns the content of the value.
+	// For primitive values, it returns the content octets.
+	// For constructed values, it returns nil because the content is
+	// the data of all members.
+	Content() []byte
+}
+
+// ConvertBERToDER converts BER-encoded ASN.1 data structures to DER-encoded.
+func ConvertBERToDER(ber []byte) ([]byte, error) {
```

Comment:
Since the package is renamed to `ber`, this function can be renamed to `ConvertToDER`. The caller would just do `ber.ConvertToDer(...)`.

Reply:
Updated